engine/src/main/java/nl/inl/blacklab/search/results/CapturedGroupsImpl.java

```diff
@@ -114,7 +114,7 @@ public Map<String, Span> getMap(Hit hit, boolean omitEmpty) {
             return null;
         Map<String, Span> result = new TreeMap<>(); // TreeMap to maintain group ordering
         for (int i = 0; i < names.size(); i++) {
-            if (!omitEmpty || groups[i].length() > 0)
+            if (groups[i] != null && (!omitEmpty || groups[i].length() > 0))
```

Comment:
Just checking, are null groups distinct from empty groups semantically? If they're the same I think this should be:

`!omitEmpty || (groups[i] != null && groups[i].length() > 0)`

Of course we'll have `omitEmpty` always set so shouldn't matter much for us, just upstream.

Reply:
I think those conditions are equivalent, we only get null elements when the flag is set.

Reply:
Good to know, thanks!